eclair-core/src/main/scala/fr/acinq/eclair/blockchain/OnChainWallet.scala

```diff
@@ -84,11 +84,8 @@ trait OnChainAddressGenerator {
    */
   def getReceiveAddress(label: String = "")(implicit ec: ExecutionContext): Future[String]
 
-  /**
-   * @param receiveAddress if provided, will extract the public key from this address, otherwise will generate a new
-   *                       address and return the underlying public key.
-   */
-  def getReceivePubkey(receiveAddress: Option[String] = None)(implicit ec: ExecutionContext): Future[PublicKey]
+  /** Generate a p2wpkh wallet address and return the corresponding public key. */
+  def getReceivePubkey()(implicit ec: ExecutionContext): Future[PublicKey]
```

Comment:
Should we go further and rename the function to make it even more explicit? What name do you think would be best?

Reply:
Yes I think the name should be explicit, since `static_remotekey` explicitly sets the `to-remote` output to a P2WPKH address and this method is only used with `static_remotekey`. Maybe `getP2wpkhPubkey()` ?
(We would also add a parameter to specify which type of key we want bitcoind to generate but we can always do that if/when we need it).

Reply:
Done in https://github.com/ACINQ/eclair/pull/2486/commits/b950b79a51c9a9e15ca0418838ceabb5dd50d79a